### PR TITLE
Component alignment script fixes

### PR DESCRIPTION
**Pull request by TomasHofman**
* don't reveal SMTP password in debugging mode,
* don't specify any default logger URL - logger won't be used if URL is
  not set,
* update usage message.

### CODE REVIEW

**Comment by TomasHofman** [comment]
Hello @rpelisse, I would like to suggest couple of changes and fixes, consider yourself.

File: upgrade-components-report.sh [hunk 1]
```diff
@@ -1,8 +1,21 @@
 #!/bin/bash
-set -eou pipefail
+set -eo pipefail
```
Comment (TomasHofman) [inline upgrade-components-report.sh:2]: I think the "pipefail" value belongs to "-o" not "-u"? But maybe it works as it was, not sure.

File: upgrade-components-report.sh [hunk 2]
```diff
@@ -1,8 +1,21 @@
...
+  echo Create following mandatory env variables:
+  echo '* TARGET_DIR - path to maven project directory which should be analyzed'
+  echo '* JOB_NAME - id which will be used to lookup a job configuration in the settings CSV file'
+  echo '* TO_ADDRESS - an email address to send the report to'
+  echo
+  echo Optional env variables:
+  echo '* JOBS_SETTINGS - path to job settings CSV file'
+  echo '* CONFIG - path to JSON configuration file with upgrade rules'
+  echo
+  echo "$(basename "${0}")"
```
Comment (TomasHofman) [inline upgrade-components-report.sh:18]: I understand we don't take any arguments anymore.

File: upgrade-components-report.sh [hunk 3]
```diff
@@ -24,10 +37,14 @@ checkEnvVar() {
...
-       -Dlogger.uri="${COMPONENT_UPGRADE_LOGGER}" \
+  local loggerParams=''
+  if [ -n "${LOGGER_URI}" ]; then
+    loggerParams="-Dlogger.projectCode="${LOGGER_PROJECT_CODE}" -Dlogger.uri=${COMPONENT_UPGRADE_LOGGER}"
+  fi
+
+  java $loggerParams \
        -jar "${CLI}" 'generate-html-report' \
-       -c "${CONFIG}" -f "${target}/" -o "${REPORT_FILE}"
+       -c "${CONFIG}" -f "${target}" -o "${REPORT_FILE}"
```
Comment (TomasHofman) [inline upgrade-components-report.sh:47]: This is to make the COMPONENT_UPGRADE_LOGGER parameter optional.

File: upgrade-components-report.sh [hunk 4]
```diff
@@ -24,10 +37,14 @@ checkEnvVar() {
...
-       -Dlogger.uri="${COMPONENT_UPGRADE_LOGGER}" \
+  local loggerParams=''
+  if [ -n "${COMPONENT_UPGRADE_LOGGER}" ]; then
+    loggerParams="-Dlogger.projectCode="${LOGGER_PROJECT_CODE}" -Dlogger.uri=${COMPONENT_UPGRADE_LOGGER}"
+  fi
+
+  java $loggerParams \
        -jar "${CLI}" 'generate-html-report' \
-       -c "${CONFIG}" -f "${target}/" -o "${REPORT_FILE}"
+       -c "${CONFIG}" -f "${target}" -o "${REPORT_FILE}"
```
Comment (TomasHofman) [inline upgrade-components-report.sh:47]: This is to make the COMPONENT_UPGRADE_LOGGER parameter optional.

File: upgrade-components-report.sh [hunk 5]
```diff
@@ -42,16 +59,14 @@ ifRequestedPrintUsageAndExit() {
 sendReportByEmail() {
   local report_file=${1}
 
-  if [ -n "${SMTP_PASSWORD}" ]; then
```
Comment (TomasHofman) [inline upgrade-components-report.sh:45]: The `if [ -n "${SMTP_PASSWORD}" ]` condition is present twice.

File: upgrade-components-report.sh [hunk 6]
```diff
@@ -75,20 +90,40 @@ deleteOldReportFile() {
...
+       -s "Possible component upgrades report - ${REPORT_TITLE}" \
+       "${TO_ADDRESS}" < "${REPORT_FILE}"
+
+  # reset debugging
+  [[ $debug == 1 ]] && set -x
+}
+
+unpackSmtpPassword() {
+  echo $(gpg -d "${GMAIL_SMTP_PASSWORD_FILE}" 2> /dev/null)
 }
```
Comment (TomasHofman) [inline upgrade-components-report.sh:127]: This is to not reveal password when run with `bash -x`.

File: upgrade-components-report.sh [hunk 7]
```diff
@@ -75,20 +90,40 @@ deleteOldReportFile() {
...
+       -e "set smtp_url = 'smtps://${FROM_ADDRESS}@smtp.gmail.com:465'" \
+       -e "set smtp_pass = '***'" \
+       -e "set ssl_starttls = yes" \
+       -e "set ssl_force_tls = yes" \
+       -s "Possible component upgrades report - ${REPORT_TITLE}" \
+       "${TO_ADDRESS}" < "${REPORT_FILE}"
+  
+  mutt -e 'set content_type = text/html' \
+       -e "set smtp_url = 'smtps://${FROM_ADDRESS}@smtp.gmail.com:465'" \
+       -e "set smtp_pass = '`unpackSmtpPassword`'" \
```
Comment (rpelisse) [inline upgrade-components-report.sh:115]: => "set smtp_pass = '$(unpackSmtpPassword`)"

;)

**Comment by rpelisse** [comment]
Sounds pretty good! Let me finish some over stuff and I'll merge it on Olympus and test it.

**Comment by TomasHofman** [comment]
For the record (also to myself), I was testing with following sample CSV content (I couldn't find the real file anywhere...):

template.csv:
```
job-name,rule-name,Wildfly - master,logger-code
```

and the command I'm running:
```
export CONFIG=../dependency-alignment-configs/rules-wildfly-master.json
export TARGET_DIR=../component-upgrade-logger
export JOB_NAME=job-name
export TO_ADDRESS=thofman@redhat.com
export JOBS_SETTINGS=template.csv

bash -x upgrade-components-report.sh
```

**Comment by rpelisse** [comment]
>For the record (also to myself), I was testing with following sample CSV content (I couldn't find the real file anywhere...):

You can't, this file is generated by Ansible ;) https://github.com/jboss-set/zeus/blob/olympus/roles/jobs/templates/component-alignment-config-template.csv.j2

**Comment by TomasHofman** [comment]
OK, I saw that info being present in some yaml config and wondered if it's a duplicity or a work in progress... Thanks :).

**Comment by spyrkob** [comment]
@TomasHofman if this is still relevant, could you rebase the PR please?

**Comment by TomasHofman** [comment]
@spyrkob rebased, please do merge this :).

Note that I'm not able to run the tests - some of the tests are failing (IMO unrelated to this) and the ones that should be testing this script are not run at all for some unknown reason.

**Pull request merged**